File: packages/discord/src/commands/categories.ts
import { Command } from "../utils/Command";
import { PREFIX } from "@daily-news-discord/environment";
import { MessageEmbed } from "discord.js";
import { categoryArray } from "@daily-news-discord/news";

export default new Command({
  name: "categories",
  description: "Shows a list of categories",
  aliases: ["cat"],
  usage: `${PREFIX}help [command]`,
  execute: async (msg) => {
    // Create a Embed of all categories
    const embed = new MessageEmbed()
      .setTitle("Categories")
      .setColor("#0099ff")
      .setFooter({ text: "Daily News" });

    embed.setDescription(categoryArray.map((c) => `\`${c}\``).join(", "));
    // Send the embed
    msg.reply({ embeds: [embed] });
  },
});

File: commands/categories.go
package commands

import (
	"strings"

	"github.com/SushiWaUmai/daily-news-discord/news"
	"github.com/bwmarrin/discordgo"
)

func init() {
	var defaultMemberPermissions int64 = discordgo.PermissionViewChannel | discordgo.PermissionSendMessages
	dmPermission := false

	createCommand(&discordgo.ApplicationCommand{
		Name:        "categories",
		Description: "prints all categories",
		DefaultMemberPermissions: &defaultMemberPermissions,
		DMPermission:             &dmPermission,
	}, func(dg *discordgo.Session, i *discordgo.InteractionCreate) {
		// Create a string of all categories
		categories := strings.Join(news.Categories, ", ")

		// Create an embed message
		embed := &discordgo.MessageEmbed{
			Title:       "Categories",
			Description: categories,
			Color:       0x0099ff,
			Footer: &discordgo.MessageEmbedFooter{
				Text: "Daily News",
			},
		}

		// Send the embed as a response to the interaction
		response := &discordgo.InteractionResponse{
			Type: discordgo.InteractionResponseChannelMessageWithSource,
			Data: &discordgo.InteractionResponseData{
				Embeds: []*discordgo.MessageEmbed{embed},
			},
		}

		dg.InteractionRespond(i.Interaction, response)
	})
}
